pkg/event/event.go

```diff
@@ -374,18 +374,29 @@ func (e *EventHandler) GetPTPState(source EventSource, cfgName string) PTPState
 }
 
 func (e *EventHandler) listenToStateRequest() {
+	var request StatusRequest
 	for {
-		select {
-		case request := <-e.statusRequestChannel:
-			if m, ok := e.data[request.CfgName]; ok {
-				for _, v := range m {
-					if v.ProcessName == request.Source {
-						request.ResponseChannel <- v.State
+		request = <-e.statusRequestChannel
+		if m, ok := e.data[request.CfgName]; ok {
+			for _, v := range m {
+				if v.ProcessName == request.Source {
+					// Attempt to respond to the channel, or
+					// timeout if nobody listening
+					select {
+					case request.ResponseChannel <- v.State:
+						goto cont
```

Comment:
`continue` didn't work here, as it continued the inner loop, while we want to continue the outer loop